File: ActivityTranslucent/src/main/java/org/chaos/demos/activitytranslucent/MainActivity.java
package org.chaos.demos.activitytranslucent;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    public static void start(Context context) {
        Intent starter = new Intent(context, MainActivity.class);
        context.startActivity(starter);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(TAG, "onCreate: "+this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause: "+this);
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop: "+this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy: "+this);
    }

    public void openNewActivity(View v) {
        start(this);
    }
}

File: GreenDaoDemos/src/main/java/org/chaos/demos/greendao/MainActivity.kt
package org.chaos.demos.greendao

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        var bill = Bill(1, true, "Test", 35.5f, System.currentTimeMillis(), "Note")
//        saveBill(bill)
//        var bill = getBill(1)
        bill.price = 233F
        bill.category = "Main"
        bill = getBill(saveBill(bill))
        Log.d(TAG, "bill = " + bill)

        bill = Bill(15, true, "Test", 35.5f, System.currentTimeMillis(), "Note")
        Log.d(TAG, "bill5 = " + bill)
        bill = getBill(saveBill(bill))

//        startActivity(Intent(this, RemoteActivity::class.java))

        text.postDelayed({
            Log.d(TAG, "bill2 = " + bill)
            bill.price = 450f
            bill = getBill(saveBill(bill))
            Log.d(TAG, "bill3 = " + bill)

            bill = getBill(1)
            bill.price = 800F
            bill.category = "Remote"
            bill = getBill(saveBill(bill))
            Log.d(TAG, "bill4 = " + bill)
        }, 3000)
    }

    private val bills: List<Bill>
        get() = DaoManager.billDao!!.loadAll()

    private fun getBill(id: Long?): Bill {
        return DaoManager.billDao!!.load(id)
    }

    private fun saveBill(bill: Bill): Long? {
        if (DaoManager.billDao!!.hasKey(bill)) {
            DaoManager.billDao!!.update(bill)
            return bill.id
        } else {
            val row = DaoManager.billDao!!.insert(bill)
            Log.d(TAG, "saveBill: row = " + row)
            return DaoManager.billDao!!.loadByRowId(row).id
        }
    }

    private fun deleteBill(id: Long?) {
        DaoManager.billDao!!.deleteByKey(id!!)
    }

    companion object {
        private val TAG = "MainActivity"
    }
}
